File: equations/create_input_data.m
intrinsic CreateInputDataFile(label::MonStgElt)
{.}
  all_data := Read("modular_curves_data.txt");
  pos := Index(all_data, label);
  eol := Index(all_data[pos..#all_data], "\n");
  substr := all_data[pos+#label..pos+eol-1];
  gens := Split(substr,"|")[2];
  genstr := Join(Split(gens, "{},"), ",");
  genstr := genstr[1..#genstr-2];
  Write("input_data/" * label, genstr : Overwrite);
  return;
end intrinsic;

File: equations/create_input_data.py
#! /usr/local/bin/sage -python

import os
import sys
# This need sto be run outside the shell, before the script
# os.system("ulimit -n 4096")
from lmfdb import db
label = sys.argv[1]
gens = db.gps_gl2zhat_test.lucky({'label' : label}, 'generators')
os.system("mkdir -p input_data")
genstr = ', '.join([str(g)[1:-1] for g in gens])
f = open("input_data/" + label, 'w')
f.write(genstr)

